File: 7-targeted-sum/targeted_sum.py
"""
Given an array of numbers and an integer target, find two unique numbers in the array 
that add up to the target value. Return an array with the indices of those two numbers, 
or "Target not found" if no two numbers sum up to the target.
The returned array should have the indices in ascending order.
"""

def find_target(arr, target):

    seen = {}

    for index, num in enumerate(arr):
        complement = target - num
    
        if complement in seen:
            return sorted([seen[complement],index])
        
        seen[num] = index   
    return "Target not found"

File: 0007-targeted-sum/targeted-sum.js
function findTarget(arr, target) {
  const seen = new Map();

    for (const [index, num] of arr.entries()) {
        const complement = target - num;
        
        if (seen.has(complement)) {
            return [seen.get(complement), index].sort((a, b) => a - b);
        }
        
        seen.set(num, index);
    }
    
    return "Target not found";
}
